src/pdiagmat.jl

```diff
@@ -2,11 +2,14 @@
 Positive definite diagonal matrix.
 """
 struct PDiagMat{T<:Real,V<:AbstractVector{T}} <: AbstractPDMat{T}
-    dim::Int
     diag::V
```

Comment:
Just removing the field means the two-argument inner constructor is broken? Can we add it but deprecate it?